### PR TITLE
Move Source Line FFI helpers

### PR DESCRIPTION
**Pull request by InnPatron**
Move the Source Line FFI helpers from parse-pyret.js to ffi.js

As far as I can tell, error data types expect a transformed source line data structure. As I am writing other error checking modules partially in Javascript, it would be useful to have these helpers available in general.

### CODE REVIEW

File: src/js/trove/ffi.js [hunk 1]
```diff
@@ -16,6 +16,7 @@
   nativeRequires: [],
   theModule: function(runtime, namespace, uri, L, Se, O, E, EQ, ERR, S, CON, /* CH, */ ED, VS) {
     var gf = runtime.getField;
+    var srcloc = runtime.getField(S, "values");
```
Comment (blerner) [inline src/js/trove/ffi.js]: This is redundant with line 25 below (`S = gf(S, "values")`)

**Comment by blerner** [comment]
This change looks ok to me.  I don't *think* it changes the dependency graph among parse-pyret, ast, srcloc, and ffi in any harmful ways, so this should be fine.

**Comment by InnPatron** [comment]
Updated.

**Comment by InnPatron** [comment]
Hmm... Actually it would make more sense if the transformation occured within the "throw error" functions. And upon further inspection, some of them actually do (ffi.js:435) (with a funciton "makeSrcloc()" in runtime.js) while some don't. So that makes me think I should potentially move "combinePyretPos()" into runtime.js and fix the rest of the error funcitons.

**Comment by blerner** [comment]
No, that won't work, because parse-pyret still needs this functionality to exist on its own.

**Comment by InnPatron** [comment]
Alright then. Anything else?

**Pull request merged**